packages/apps/virtual-machine/templates/vm-update-hook.yaml

```
@@ -0,0 +1,118 @@
+    {{- $needUpdateType = true -}}
+  {{- end -}}
+{{- end -}}
+
+{{- if and $existingVM $instanceProfile -}}
+  {{- if not (eq $existingVM.spec.preference.name $instanceProfile) -}}
+    {{- $needUpdateProfile = true -}}
+  {{- end -}}
+{{- end -}}
+
+{{- if and $existingPVC $desiredStorage -}}
+  {{- $currentStorage := $existingPVC.spec.resources.requests.storage | toString -}}
```

yamllint (1.35.1)

[warning] wrong indentation: expected 0 but found 2

(indentation)